core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/AsyncZippedStream.scala

```diff
@@ -0,0 +1,17 @@
+package com.lynbrookrobotics.potassium.streams
+
+class AsyncZippedStream[A, B](master: ExpectedPeriodicity) extends Stream[(A, B)] {
+  override val expectedPeriodicity: ExpectedPeriodicity = master
+
+  private[this] var followerSlot: Option[B] = None
+
+  def receiveMaster(masterValue: A): Unit = {
+    if (followerSlot.isDefined) {
+      publishValue((masterValue, followerSlot.get))
+    }
+  }
+
+  def receiveFollower(followerValue: B): Unit = {
+    followerSlot = Some(followerValue)
+  }
+}
```

Comment:
Does not have strong reference to parent streams?
`data.zip(otherData.map(...))`
will break stuff cause there is are no strong references to `otherData.map(...)`